osu.Game.Tournament/TournamentGameBase.cs

```diff
@@ -150,7 +150,9 @@ private bool addPlayers()
             {
                 foreach (var p in t.Players)
                 {
-                    if (string.IsNullOrEmpty(p.Username) || p.Statistics?.GlobalRank == null)
+                    if (string.IsNullOrEmpty(p.Username)
+                        || p.Statistics?.GlobalRank == null
```

Comment:
aren't these nullable? couldn't this trigger a repopulation on every startup?

Reply:
Yeah that's one down of this solution, which happened to `GlobalRank` as well. I would say it's unlikely to add users who don't have global ranks in tournament teams logically. 

But I believe that's not a valid reason, one alternative solution would be a custom JSON converter for deserializing user that would aim at reading from `rank` object and migrate to the new outer properties.

Reply:
Let's just go with this for now. I agree it will probably not be a common occurrence, if ever.